### DISCUSSION TITLE
entitiesTs files can't be found

### DISCUSSION THREAD

**Discussion by JustDenP** [mikro-orm/mikro-orm#4204] [Q&A]
Can someone please explain what the problem is, because I have not been able to get rid of the error for several hours 

Before now I had a normal project with one folder and everything worked fine, now I'm switching to monorepo and put my entities in a separate package and trying to specify the path to it inside mikro-orm.config.ts

Now my structure looks like this:

apps
--api
packages
--shared
----models
dist
--apps...
--packages...

Inside dist/packages/shared/models/src/*.entity.js  
I have all my compiled js entities and entities.d.ts alongside

inside packages/shared/models/src/*.entity.ts 
I have all my source entities

So my config looks like this:
```ts
type: 'postgresql',
    host: process.env.DB_HOST,
    port: Number(process.env.DB_PORT),
    password: process.env.DB_PASSWORD,
    user: process.env.DB_USER,
    dbName: process.env.DB_DATABASE,
    entities: [`../../dist/packages/shared/models/src/*.entity.js`],
    entitiesTs: [`../../packages/shared/models/src/*.entity.ts`],
    loadStrategy: LoadStrategy.JOINED,
    highlighter: process.env.NODE_ENV == 'development' ? new SqlHighlighter() : undefined,
    debug: true,
    logger: logger.log.bind(loggerType),
    metadataProvider: TsMorphMetadataProvider,
    namingStrategy: UnderscoreNamingStrategy,
    entityRepository: BaseRepository,
    migrations: {
      tableName: 'migrations', // name of database table with log of executed transactions
      path: 'dist/migrations', // path to the folder with migrations
      pathTs: 'src/migrations', // path to the folder with TS migrations
      // (if used, we should put path to compiled files in `path`)
      glob: '!(*.d).{js,ts}', // how to match migration files (all .js and .ts files, but not .d.ts)
      transactional: true, // wrap each migration in a transaction
      disableForeignKeys: true, // wrap statements with `set foreign_key_checks = 0` or equivalent
      allOrNothing: true, // wrap all migrations in master transaction
      dropTables: true, // allow to disable table dropping
      safe: false, // allow to disable table and column dropping
      snapshot: true, // save snapshot when creating new migrations
      emit: 'ts', // migration generation mode
    },
  };
```
 

the "entities" works fine and all my entities are found correctly
`processing 4 files. processing entity OTP`

But entitiesTs constantly shows me the error:
`Source file '../../dist/packages/shared/models/src/otp.entity.ts' not found. Check your 'entitiesTs' option and verify you have 'compilerOptions.declaration' enabled in your 'tsconfig.json'`

Whatever paths and wildcards I use, my source entities are not loaded.
Maybe anybody knows how to fix it

Even with this syntax the same error for me
```ts
entities: [`../../dist/packages/shared/models/src/*.entity.js`],
entitiesTs: [OTP, User, UserProfile, UserSettings],  <-- my ts entities
```

```
ERROR [ExceptionHandler] Source file '../../dist/packages/shared/models/src/otp.entity.ts' not found. Check your 'entitiesTs' option and verify you have 'compilerOptions.declaration' enabled in your 'tsconfig.json'.
```

Also, I have declaration: true in my tsconfig and
```
"mikro-orm": {
  "useTsNode": true,
  "configPaths": [
    "./src/mikro-orm.config.ts",
    "../../dist/apps/api/src/mikro-orm.config.ts"
  ]
}
```
in my package.json

**Comment by B4nan** [comment]
Both `entities` and `entitiesTs` should be aligned, defined in the very same way. The way you setup the paths seems quite wrong, you should have the dist folder at the package level, not in root. But that itself is probably not the culprit. 

The fact that it complains about `entitiesTs` and prints a path with dist means you are probably not running the app is ts-node mode at all. The way you set the config paths is definitely wrong, you should have there both js and ts files. 

Provide complete repro if you want more help, or check out one of the example apps, they all work correctly.

**Comment by JustDenP** [comment]
Thank you, but I run my nestjs application, all modules are loaded before the mikroorm entity is loaded, so I don't think it's the ts-node problem

I had no experience in creating a monorepo before, so I took the settings from NX as a basis and copied, only one dist folder in root is created there too, so I thought that was the standard approach

I created a repository for more clarity
https://github.com/JustDenP/monorepo.git

I start my project with `npm run start:dev` inside apps/api

**Reply by B4nan** [reply]
It is far away from standard, i dont think i ever saw such setup. It doesnt make much sense to me, why have a monorepo when you treat it as a single package. But i havent used nx myself, idk about that. 

I tool a brief look at that repo, and there are way too many things that are wrong. You dont even install any driver anywhere? You definitely cant depend on the orm packages in multiple packages, you need a single one, rest needs to be peer deps. I found the core package installed in 3 places, that wrong, you want a single install. 

Havent really looks at the configs, I wont have time for that till next week. What i can suggest right now - simplify, your repro is far away from minimal. Start with the bare bones. That way you often find the problem yourself. And check existing issues, maybe you will find some examples in there. 

Final suggestion, try the CLI debug command, it should help exactly with those problems. The paths feel wrong, but it depends on where you run the command from.

**Reply by JustDenP** [reply]
Okay, thanks, I guess it really is easier to redo and configure everything from scratch x)

**Reply by B4nan** [reply]
It will definitely help you understand the detail, battling a ready made setup that is not really working can be really hard and painful.
